components/match2/commons/match_group_input.lua

```diff
@@ -531,10 +531,11 @@ function MatchGroupInput.fetchStandaloneMatch(matchId)
 	end)
 end
 
+---Warning, mutates first argument by removing the key `opponentX` where X is the second argument
 ---@param match table
 ---@param opponentIndex integer
 ---@param options readOpponentOptions
----@return table?
+---@return standardOpponent?
```

Comment:
it doesn't return standard opponent
it returns match2opponent plus some additional stuff that was in the opponentInput from before calling the function

Reply:
mergeRecordWithOpponent converts it into a standardOpponent (with some extra fields) no?

Reply:
until it gets passed to `mergeRecordWithOpponent` it is basically a standardOpponent plus additional fields
but `mergeRecordWithOpponent` makes it a match2opponent with additional fields
keys all lower cased, name instead of pageName etc pp